packages/forms/src/widgets/DatalistWidget/DatalistWidget.js

```diff
@@ -227,6 +252,28 @@ class DatalistWidget extends React.Component {
 		return [];
 	}
 
+	getDropdownItems(suggestions) {
+		// options with categories
+		if (this.props.options && this.props.options.groupBy) {
+			const category = this.props.options.groupBy;
```

Comment:
I guess the category const could be defined before the if statement to be used in it.

Reply:
yes, but the category const is used only inside the first if branch:)